README.md

```diff
@@ -1,28 +1,153 @@
+BADGER_ADDRESS=https://testnet3-badger.mysterium.network
+BROKER_URL=nats://testnet3-broker.mysterium.network
+UNIVERSE_JWT_SECRET=Some_Secret
+REDIS_ADDRESS=redis:6379
+REDIS_DB=0
+REDIS_PASS=
 ```
 
-Or using docker:
+##### Sidecar
 
+```bash
+MYSTERIUM_LOG_MODE=json
+QUALITY_ORACLE_URL=https://testnet3-quality.mysterium.network
+BROKER_URL=nats://testnet3-broker.mysterium.network
+REDIS_ADDRESS=redis:6379
+REDIS_DB=0
+REDIS_PASS=
+GECKO_URL=http://wiremock:8080
+COINRANKING_URL=http://wiremock:8080
+COINRANKING_TOKEN=Some_Token
 ```
-docker-compose up
+
+#### DB Tables
+
+```sql
+CREATE TABLE IF NOT EXISTS country_price_multipliers(
+   id                SERIAL PRIMARY KEY,
+   country_code      varchar(2) not null UNIQUE,
+   multiplier        double precision NOT NULL,
+   updated_at        timestamp NOT NULL default now()
+);
+CREATE TABLE IF NOT EXISTS pricing_config(
+   id                SERIAL PRIMARY KEY,
+   cfg   jsonb       NOT NULL,
+   created_at        timestamp NOT NULL default now()
+);
+CREATE TABLE IF NOT EXISTS proposals
+(
+   key               text PRIMARY KEY,
+   proposal          jsonb NOT NULL,
+   expires_at        timestamp NOT NULL,
+   COLUMN updated_at timestamp NOT NULL
+);
 ```
 
+#### DB Migrations
+
+Path for migration files: `db/migrations`
+
+All migrations are done on Discovery service start up
+
+#### NATS Msg Broker channels
+
+* `*.proposal-register.v3` - Register new Node proposal
+* `*.proposal-ping.v3` - Node Ping
+* `*.proposal-unregister.v3` - Unregister Node proposal
+
+#### Entries
+
+* `cmd/main.go` - Discovery service
+* `sidecar/cmd/main.go` - Sidecar Pricing Parser service
+
+#### Code structure
+
+* `/config` - [Discovery] config parser. Env params
+* `/tags` - [Discovery] Tags proposals enhancer for SuperProxy 
+* `/docs` - [Discovery] Auto generated Swagger for REST API
+* `/e2e` - e2e tests
+* `/health` - [Discovery] health checker REST API 
+* `/listener` - [Discovery] NATS listener
+* `/price/api.go` - [Discovery] Pricing REST API
+* `/price/config.go` - [Discovery, Sidecar] Pricing config
+* `/price/market.go` - [Sidecar] Scheduled price fetcher from different APIs (Gecko, Coinmarket) 
+* `/price/network_load_calculator.go` - [Sidecar] Updating countries multipliers based on load (sessions/providers)
+* `/price/price_getter.go` - [Discovery] Fetch prices from Redis
+* `/price/price_updater.go` - [Sidecar] Scheduled price updater to Redis
+* `/proposal/metrics/echnancer.go` - [Discovery] Proposal quality enhancer
+* `/proposal/v3` - [Discovery] Proposal structs
+* `/proposal/api.go` - [Discovery] Proposal REST API
+* `/proposal/repository.go` - [Discovery] Mappings for Postgres
+* `/proposal/service.go` - [Discovery] Proposal service with scheduled expiration job
+* `/quality/oracleapi` - [Discovery] Quality Oracle REST API client
+* `/quality/service.go` -  [Discovery] Caching layer with BigCache for Quality Oracle responses
```

Comment:
IMO this is very nice but perhaps we should reduce the scope to package's as oppose to individual files? In other words unless we have some pre-commit hook to validate if all is up-to date this is going to get stale more often than not.

Reply:
The problem that structure not clear so i cant make package scope clear. Any way it's dev mandatory to updated documentation. In other way we will have problems on all levels.